checkUrlExists/lambda/index.js

```diff
@@ -50,10 +40,50 @@ async function checkUrl(url) {
 }
 
 async function run(url) {
-  const curlOpts = {
-    caInfo: certFilePath,
-  };
-  return curly.head(url, curlOpts);
+  const parsedUrl = Url.parse(url);
+  const protocol = parsedUrl.protocol; // Url.parse() lower cases the protocol
+  if (FTP_PROTOCOL === protocol || SECURE_FTP_PROTOCOL === protocol) {
+    const secure = SECURE_FTP_PROTOCOL === protocol;
+    const ftpClient = new ftp.Client();
+    try {
+      let options = {
+        host: parsedUrl.host,
+        secure: secure,
+      };
+      if (parsedUrl.port) {
+        options = { port: parsedUrl.port, ...options };
+      }
+      await ftpClient.access(options);
+      const size = await ftpClient.size(parsedUrl.path);
+      return size > 0
```

Comment:
Is it plausible that a valid file will be empty?  If so, `>=` here?

Reply:
Hmm, I was thinking an empty file would never be valid from a workflow input perspective. But I could see it both ways. I'm going to leave it as is for now, but if you or anybody feels strongly, I can change it.